pandas/tools/merge.py

```diff
@@ -1186,6 +1203,11 @@ def flip(xs):
             if tolerance is not None:
                 tolerance = tolerance.value
 
+        # enumeration for direction
```

Comment:
I'll try your suggestion above of having three different functions for the search instead of one function with giant conditional branches. That means I can just use `_asof_function()` to get the attribute with the direction string.